### DISCUSSION TITLE
What is the correct response to a Subscribe message?

### DISCUSSION THREAD

**Discussion by blevine** [enisdenjo/graphql-ws#600] [Q&A] [answered]
The protocol doesn't state what to send in response to a Subscribe message. Should the query be executed immediately and the response by a Next message with the result?  Or should their be no response message at all until the subscription event fires and is returned in a Next message?

**Comment by enisdenjo** [comment, marked as answer]
The query should execute immediately. Next messages are sent when results are ready.
